### PR TITLE
Implements connection check while adding datasource

### PR DESCRIPTION
**Pull request by YaroslavTir**
move check functionality from enterprise

### CODE REVIEW

File: src/main/resources/application-base.yml [hunk 1]
```diff
@@ -122,6 +123,7 @@ proxy:
     username:
     password:
 executionEngine:
+  host: qa.odysseusovh02.odysseusinc.com
```
Comment (acumarav) [inline src/main/resources/application-base.yml]: Should it be in the base?

File: src/main/java/com/odysseusinc/arachne/datanode/Constants.java [hunk 2]
```diff
@@ -119,7 +119,8 @@
             String DELETE = "/api/v1/data-sources/{id}";
             String UPDATE = "/api/v1/data-sources/{id}";
             String CENTRAL_REGISTER = "/api/v1/data-sources/{id}/register-on-central";
-            String DS_MODEL_CHECK_RESULT = "/api/v1/data-sources/{id}/check/result/{password}";
+            String DS_MODEL_CHECK_FIRSTCHECK ="/api/v1/data-sources/{id}/check/result/{password}/firstcheck";
+            String DS_MODEL_CHECK_HEARTBEAT = "/api/v1/data-sources/{id}/check/result/{password}/heartbeat";
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/Constants.java]: Where it is used?

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java [hunk 3]
```diff
@@ -319,7 +355,8 @@ public void updateHealthStatus(Long centralId, CommonHealthStatus status, String
     @Override
     public AutoDetectedFields autoDetectFields(DataSource dataSource) {
 
-        return new AutoDetectedFields(CommonModelType.CDM);
+        CommonDataSourceDTO dataSourceDTO = checkDataSource(dataSource);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java]: This wouldn't be called in Standalone mode

File: src/main/java/com/odysseusinc/arachne/datanode/Constants.java (hunk 2 above)
Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/datanode/Constants.java]: Pls remove any related to heartbeat or healthcheck from public

File: src/main/resources/application-base.yml (hunk 1 above)
Comment (YaroslavTir) [reply src/main/resources/application-base.yml]: fixed

File: src/main/java/com/odysseusinc/arachne/datanode/Constants.java (hunk 2 above)
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/Constants.java]: fixed

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java (hunk 3 above)
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java]: yeah, this is right, but as I can see we call this only for NETWORK 
```
    public void createOnCentral(User owner, DataSource dataSource) {

        if (Objects.equals(NETWORK, dataNodeService.getDataNodeMode())) {
            AutoDetectedFields autoDetectedFields = autoDetectFields(dataSource);
```

Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java]: fixed

**Comment by wivern** [comment]
Failed to edit existing datasource.
Steps to reproduce:
1. Open an existing data source
1. Press save without any changes
1. Got an error "Data source name is not unique"

**Pull request merged**